scrunch/expressions.py

```diff
@@ -475,20 +475,53 @@ def category_ids(var_id, var_value, variables=variables):
                 return var_value
             return value
 
+        def adapt_multiple_response(var_url, values):
+            # if variable is multiple_response
+            #       convert value --> column and change ids by aliases
+            var_id = variable_id(var_url)
+            aliases = []
+            for var in variables:
+                if variables[var]['id'] == var_id:
+                    for sub_var_id in variables[var]['subreferences']:
+                        # man this is deep dictionary
+                        aliases.append((
+                            sub_var_id,
+                            variables[var]['subreferences'][sub_var_id]['alias']))
+            cat_to_ids = [
+                tup[0] for tup in aliases if int(tup[1].split('_')[-1]) in values]
+            return [{'variable': var_url}, {'column': cat_to_ids}], False
+
+        def is_multiple_response(var_url):
+            # make sure the variable is multiple response
+            var_id = variable_id(var_url)
+            for var in variables:
+                if variables[var]['id'] == var_id:
+                    if variables[var]['type'] == 'multiple_response':
```

Comment:
Isn't the type available straight from the variable? `var_resource.body.type` instead of iterating? Consider that datasets can have multiple tens of thousands of variables, iterating over all variables should be considered an expensive operation.